tests/test_A0_initialization.py

```diff
@@ -8,79 +8,92 @@
-            assert not os.path.exists(os.path.join(self.output_path, "dummy"))
-
-    def test_create_output_path_existing_yes(self):
-        initializing.check_output_directory(
-            os.path.join(self.output_path), overwrite=True
-        )
-        assert os.path.exists(self.output_path)
-        assert not os.path.exists(os.path.join(self.output_path, "dummy"))
-
-    def teardown_method(self):
-        shutil.rmtree(self.output_path, ignore_errors=True)
 
-
-class TestUserInput:
+class TestProcessUserArguments:
 
     output_path = os.path.join(".", "tests", "MVS_outputs")
     input_path = os.path.join("tests", "inputs")
     fake_input_path = os.path.join("tests", "fake_inputs")
 
-    def test_user_input_path_folder_copy_in_output(self):
-        initializing.get_user_input(path_output_folder=self.output_path)
-        assert os.path.exists(self.output_path)
-
-    def test_user_input_path_input_folder_not_existing(self):
-        with pytest.raises(NotADirectoryError):
-            initializing.get_user_input(
-                path_input_file=self.fake_input_path,
-                path_output_folder=self.output_path,
-            )
-
-    def test_user_input_path_input_file_not_existing(self):
-        with pytest.raises(FileNotFoundError):
-            initializing.get_user_input(
-                path_input_file=os.path.join(self.input_path, "not_existing.json"),
-                path_output_folder=self.output_path,
-                overwrite=True,
-            )
+    def test_input_folder_is_copied_in_output_within_folder_named_input(self):
+        pass
+        # initializing.get_user_input(path_output_folder=self.output_path)
+        # assert os.path.exists(self.output_path)
+
+    def test_input_folder_not_existing_raise_filenotfound_error(self):
+        pass
+        # with pytest.raises(NotADirectoryError):
+        #     initializing.get_user_input(
+        #         path_input_file=self.fake_input_path,
+        #         path_output_folder=self.output_path,
+        #     )
+
+    def test_if_json_opt_and_no_json_file_in_input_folder_raise_filenotfound_error(
+        self,
+    ):
+        pass
+        # with pytest.raises(FileNotFoundError):
+        #     initializing.get_user_input(
+        #         path_input_file=os.path.join(self.input_path, "not_existing.json"),
+        #         path_output_folder=self.output_path,
+        #         overwrite=True,
+        #     )
+
+    def test_if_json_opt_and_more_than_one_json_file_in_input_folder_raise_fileexists_error(
+        self,
+    ):
+        pass
+        # with pytest.raises(FileNotFoundError):
+        #     initializing.get_user_input(
+        #         path_input_file=os.path.join(self.input_path, "not_existing.json"),
+        #         path_output_folder=self.output_path,
+        #         overwrite=True,
+        #     )
+
+    def test_if_json_opt_path_input_file_set_to_path_input_folder(self):
```

Comment:
I would say that we can remove parameter `path_input_file ` and replace it by 

for json

    file_path = path_input_folder + name_json + input_type

or for csv input

    file_path =  path_input_folder + ´csv_elements´ + ´economic_parameters´ + input_type

or for sequences

    file_path = path_input_folder + dict[´energyProduction´][´timeseries´][´file_name´]

...in the end, all files in the input folder are input files, and it may be confusing...?

Reply:
Ok, then we'll have the tests checking that the expected files built from the given folder path are present for each of these 2 csv/json situation if the respective `input_type` are chosen. The sequence test should run regardless of which of csv/json option was chosen, no?

Reply:
> expected files built from the given folder path are present for each of these 2 csv/json situation

Not sure what you mean by that...?

I agree, the sequences in csv format should always be run. I did not see the respective test, though.

Reply:
if opt json is chosen, there should be a file under `path_input_folder + name_json +".json"`

`name_json` should always be the same as the user only input the `path_input_folder`, I suggested `mvs_config.json`

if opt csv is chosen, there should be a file under `path_input_folder + "csv_elements" + "economic_parameters.csv"`
(To be sure, the path would be `<path_input_folder>/csv_elements/economic_parameters.csv`)

Reply:
> if opt json is chosen, there should be a file under `path_input_folder + name_json +".json"`

+1

> `name_json` should always be the same as the user only input the `path_input_folder`, I suggested `mvs_config.json`

+1 

> if opt csv is chosen, there should be a file under `path_input_folder + "csv_elements" + "economic_parameters.csv"`
> (To be sure, the path would be `<path_input_folder>/csv_elements/economic_parameters.csv`)

I wrote "economic_parameters.csv" only as an example. Necessary are these files:

    required_files_list = [
        "energyConsumption",
        "simulation_settings",
        "project_data",
        "economic_data",
    ]

Hard-coded [here](https://github.com/rl-institut/mvs_eland/blob/dev/src/A1_csv_to_json.py), lines 68ff.


Or we do not allow people to delete any csv - all allowed files must be included, and if necessary not have any columns wiht entries. List of these files [here](https://github.com/rl-institut/mvs_eland/blob/dev/src/A1_csv_to_json.py), ll. 23.

    ALLOWED_FILES = (
        "fixcost",
        "simulation_settings",
        "project_data",
        "economic_data",
        "energyConversion",
        "energyProduction",
        "energyStorage",
       "energyProviders",
        "energyConsumption",
    )

Reply:
"fixcost" is not used jet and could also be deleted (for now).

Reply:
I think these should be stored in variables in a module under `src/constants.py` as I suggested [here](https://github.com/rl-institut/mvs_eland/pull/141/files#r386474222)

Reply:
True. But do we want to differ between ALLOWED_FILES and NECESSARY_FILES? 

There should also be a test if there are other files then the ALLOWED_FILES in csv_elements - these are not used for the simulation and should produce an error code.

Reply:
I propose to require the user to place all ALLOWED_FILES into the csv_elements folder.